server/openai/generateItenerary.js

```diff
@@ -0,0 +1,56 @@
+const openaiClient = require('./openaiClient')
+
+async function generateItinerary(constraints) {
+  try {
+    const conversation = [
+      {
+        role: 'system',
+        content: `You are an AI that generates travel itineraries. Respond with ONLY JSON that contains the plan for each stage. 
+        ONLY respond with the following format, do not include any descriptions or codeblocks, I should be able to parse the output to a JavaScript Object
+        The response needs to be formatted exactly like the following structure. Add some Food and snack stages as well with an optional: true marker.:
+        '''
+        {
+          plan:[
+            {
+              day: Number
+              bullet_point_summary: String (comma seperated. Can be the names of each place)
+              stages:[
+                {
+                  stage: Number,
+                  place: String,
+                  description: String,
+                  address: String,
+                  emoji: String (best emoji representation of stage)
+                }
+              ]
+            }
+          ]
+        }
+        '''
+        
+        If there are any confusing values respond with:
+        '''
+        {
+          error: String (reason)
+        }
+        '''
+        `,
+      },
+      {
+        role: 'user',
+        content: `Generate a travel itinerary based on these constraints: ${constraints}`,
+      },
+    ]
+
+    const response = await openaiClient(conversation)
+    if (response) {
+      return response
```

Comment:
```suggestion
      if (response.error) throw {name : "BadRequestError", message : response.error}; 
      return response
```
Relating to the below conversation, should we handle the Romeo case here too?

Reply:
btw idk if that throw will actually work, just took it from here https://stackoverflow.com/a/871646

Reply:
or doing this might b better
https://stackoverflow.com/a/44549774

however if u think its too complicated, can just consider writing a comment saying smth like: 
`// response either contains trip data, or a message explaining why the initial request was invalid`

Reply:
The original response type of this function was meant to be `String` so I think we respond just as is and let the backend handle conversion to JS Object.